script.sublissimo/resources/lib/menu_items/syncing.py

```diff
@@ -0,0 +1,84 @@
+import xbmc
+import xbmcgui
+import xbmcaddon
+import os
+import sys
+
+from ..players.player_controller import PlayerInstance
+from .. import script
+
+ADDON = xbmcaddon.Addon()
+__addon__     = xbmcaddon.Addon()
+_  = __addon__.getLocalizedString
+
+def create_menu(subtitle):
+    initiallayout = [(_(32093), "videodb://movies/titles/"),
+                          (_(32094), "videodb://tvshows/titles/"),
+                          (_(32095), "videodb://"),
+                          (_(32005), "")]
+    if subtitle.videodbfilename:
+        initiallayout.insert(0, (_(32116), subtitle.videodbfilename))
+    if subtitle.videofilename:
+        base = os.path.basename(subtitle.videofilename)
+        initiallayout.insert(0, (_(35008) + base, subtitle.videofilename))
+    return initiallayout
+
+def retrieve_video(subtitle):
+    initiallayout = create_menu(subtitle)
+    choice = xbmcgui.Dialog().contextmenu([strings for strings, locations in initiallayout])
+    if choice == -1 or initiallayout[choice][1] == "":
+        script.show_dialog(subtitle)
+    pos_locations = [locations for strings, locations in initiallayout]
+    if pos_locations[choice] in (subtitle.videofilename, subtitle.videodbfilename):
+        return pos_locations[choice]
+    location = xbmcgui.Dialog().browse(
+                  1, _(32020), 'video', '', False, False, pos_locations[choice])
+    if location in pos_locations or not location:
+        script.show_dialog(subtitle)
+    if not subtitle.videofilename:
+        subtitle.videodbfilename = location
+    return location
+
+def check_player_instances(subtitle=None):
+    PlayerInstance().deactivate()
+    if subtitle:
+        subtitle.delete_temp_file()
+
+def experiment(typeofinstance, subtitle):
+    if xbmc.Player().isPlayingVideo():
+        if xbmc.Player().getPlayingFile() == subtitle.videofilename:
+            player = PlayerInstance().request(typeofinstance)
+            player.add(subtitle)
+            player.start()
+            xbmc.Monitor().waitForAbort()
```

Comment:
please use a default sleep time otherwise you'll be stuck here until kodi exits

Reply:
fixed

Reply:
Sorry this is wrong. I meant `xbmc.Monitor().waitForAbort(1)` instead of adding `xbmc.sleep(1000)`

Reply:
Whoops, fixed